github_pr.py

```diff
@@ -179,19 +179,38 @@ def _check_owner_cannot_ship(owner, comment_users):
 
 
 def _merge_pr(pr, **args):
-    if args['mergecondition_approved_mergers'] or args['mergecondition_approved_mergers_file'] or args['mergecondition_owner_cannot_ship']:
+    if args['condition_approved_mergers'] or args['condition_approved_mergers_file'] or args['condition_owner_cannot_ship']:
         merge_comment_users = [comment.user.login for comment in _load_issue(**args).get_comments().reversed if re.search(".*%s.*" % args['mergecomment'], comment.body)]
-        if args.get('mergecondition_approved_mergers'):
-            merge_comment_users = _check_approved_mergers(args['mergecondition_approved_mergers'], merge_comment_users)
-        if args.get('mergecondition_approved_mergers_file'):
-            merge_comment_users = _check_approved_mergers_file(args['mergecondition_approved_mergers_file'], merge_comment_users)
-        if args.get('mergecondition_owner_cannot_ship'):
+        if args.get('condition_approved_mergers'):
+            merge_comment_users = _check_approved_mergers(args['condition_approved_mergers'], merge_comment_users)
+        if args.get('condition_approved_mergers_file'):
+            merge_comment_users = _check_approved_mergers_file(args['condition_approved_mergers_file'], merge_comment_users)
+        if args.get('condition_owner_cannot_ship'):
             merge_comment_users = _check_owner_cannot_ship(pr.user.login, merge_comment_users)
         if merge_comment_users:
             pr.merge()
     else:
         pr.merge()
 
+def github_check_condition(**args):
+    check_required_fields(['token', 'repo', 'number'], **args)
+    gh = Github(args['token'])
+    repo = gh.get_repo(args['repo'])
+    print "PR NUMBER", args['number']
+    pr = repo.get_pull(args['number'])
+    issue = _load_issue(**args)
+    merge_comment_users = [comment.user.login for comment in issue.get_comments().reversed if re.search(".*%s.*" % args['mergecomment'], comment.body)]
```

Comment:
esp. if we're going to need to deal with timestamps and making sure the valid mergecomment is on the latest sha of the PR, I think this moving to its own function makes more sense